src/GitHub_Updater/Rest_Update.php

```diff
@@ -248,15 +294,17 @@ public function process_request() {
 	private function get_local_branch() {
 		if ( isset( $_REQUEST['plugin'] ) ) {
 			$repos = Singleton::get_instance( 'Plugin', $this )->get_plugin_configs();
-			$repo  = $repos[ $_REQUEST['plugin'] ];
+			$repo  = isset($repos[ $_REQUEST['plugin'] ]) ? $repos[ $_REQUEST['plugin'] ] : null;
 		}
 		if ( isset( $_REQUEST['theme'] ) ) {
 			$repos = Singleton::get_instance( 'Theme', $this )->get_theme_configs();
-			$repo  = $repos[ $_REQUEST['theme'] ];
+			$repo  = isset($repos[ $_REQUEST['theme'] ]) ? $repos[ $_REQUEST['theme'] ] : null;
```

Comment:
This is also inside a conditional of the same test.

Reply:
You're definitely right, maybe due to the following: `if (isset ($ repo))` (and night developing...)

Reply:
and maybe we both made a mistake...

> if ( *isset( **$_REQUEST['theme']***) ) {
> ...
> $repo  = *isset(**$repos**[ $_REQUEST['theme'] ])* 
                   ? $repos[ $_REQUEST['theme'] ]
                   : null;
> }

Reply:
Obviously it is a warning only if the theme name is wrong / non-existent

Reply:
similarly for `if (isset ($ repo))`

Reply:
The latest version has improvements to where $repo is defined as false initially and if false, the $current_branch is set to 'master'.

https://github.com/afragen/github-updater/blob/e5f027b4e6a78f6ee475a60799d8308ba4d0ada5/src/GitHub_Updater/Rest_Update.php#L256-L271

Reply:
but `$repos[ $_REQUEST['theme/plugin name'] ]; ` still generates a log warning if a wrong name is supplied, am i wrong?

---
eg. Wrong theme name = "test-wrong"

[07-Apr-2018 21:33:52] PHP Notice:  Undefined index: test-wrong in "..."/plugins/github-updater/src/GitHub_Updater/Rest_Update.php on line 264

---
indeed, a you correctly pointed before, if no theme name is supplied it will never enter into the `if ( isset( $_REQUEST['theme'] ) )` condition

Reply:
You are correct. https://github.com/afragen/github-updater/commit/2e1eb1328881325493bf5ab5bab6aeea66fb19d6 should fix this.